### PR TITLE
Migrate to Expo

### PR DESCRIPTION
**Pull request by lucasloisp**
Migrates the app that into Expo, following its most basic template (`create-expo-app`) which includes [Expo Router](https://docs.expo.dev/router/introduction/). Most of the screens are moved to satisfy this requirement, as Expo Router uses a file-based structure for building navigators. Aside from that, the intent was to preserve as much of the existing file structure as possible.

Most of the dependencies were fresh installed, and thus bumped — we're now on prettier `3.3.1`, and all files were reformatted to match (mostly semicolon related, nothing too fancy). Another important note is that some files were removed, as they were unused. Some of that code was about `.svg` imports, which I did not find in use, but I might be missing context and perhaps they were intended as a part of one of the challenges. 

The goal with this migration is simplifying the setup necessary by those trying to use Slimy to learn. By using the standard Expo template, along with commonplace dependencies, we're able to run Slimy on Expo Go, which removes the need for any native setup or compilation step, reducing the setup to "install and start", plus scanning a QR code. Additionally, relying on Expo arguably makes the code more direct, as the vast majority is now directly related to Slimy and its animations.

Functionality wise, there are no (intended) changes to Slimy or any of the flows.

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "jsxBracketSameLine": true,
+  "bracketSameLine": true,
```
Comment (lucasloisp) [inline .prettierrc:2]: `jsxBracketSameLine` was [deprecated](https://prettier.io/docs/en/options.html#deprecated-jsx-brackets) in `prettier@2.4.0` in favor of `bracketSameLine`.

File: babel.config.js [hunk 2]
```diff
@@ -1,12 +1,6 @@
-module.exports = {
-  presets: [
-    [
-      'module:metro-react-native-babel-preset',
-      { useTransformReactJSXExperimental: true },
-    ],
-  ],
-  plugins: [
-    ['@babel/plugin-transform-react-jsx', { runtime: 'automatic' }],
-    'react-native-reanimated/plugin',
```
Comment (lucasloisp) [inline babel.config.js:10]: The Reanimated plugin is [automatically configured](https://docs.expo.dev/versions/latest/sdk/reanimated/#installation) in `babel-preset-expo`

File: metro.config.js [hunk 3]
```diff
@@ -1,16 +0,0 @@
-const { getDefaultConfig } = require("metro-config");
-
-module.exports = (async () => {
-  const {
-    resolver: { sourceExts, assetExts }
-  } = await getDefaultConfig();
-  return {
-    transformer: {
-      babelTransformerPath: require.resolve("react-native-svg-transformer")
```
Comment (lucasloisp) [inline metro.config.js:9]: I found no `.svg` file imports on any of the solutions. Is it still perhaps an intended part of the "work in progress" and therefore something we want to support?

File: src/app/solutions/belus.tsx [hunk 4]
```diff
@@ -254,3 +258,5 @@ export const BelusChallenge: React.FC<BelusChallengeProps> = ({}) => {
     </Animated.View>
   )
 }
+
+export default BelusChallenge
```
Comment (lucasloisp) [inline src/app/solutions/belus.tsx:260]: Expo Router uses `default` exports by convention to detect Screens.

File: src/assets/mouth.svg [hunk 5]
```diff
@@ -1,4 +0,0 @@
-<svg width="60" height="30" viewBox="0 0 60 30" fill="none">
```
Comment (lucasloisp) [inline src/assets/mouth.svg]: Noticed these SVGs are set up as animated Components on each solution (e.g. `Slimy/Mouth/index.tsx`).

Should we keep the source `.svg` as a reference for students? Perhaps in some sort of `reference/` folder. Or, together with the `svg-transformer`, is it the intention that students start out importing the `svg` file and move on to the component version once it's animated?

File: metro.config.js (hunk 3 above)
Comment (carozo) [reply metro.config.js:9]: If I remember correctly, I used this while I was drawing the SVGs to be able to preview them on Slimy. But I think we can remove it, as there is no real need to have this library — there are many other SVG external tools that can serve the same purpose.

File: src/assets/mouth.svg (hunk 5 above)
Comment (carozo) [reply src/assets/mouth.svg]: Hmm, that's a good question. I don't think it is necessary to keep the `svg-transformer` library, but the `.svg` files are necessary for them to be able to copy and animate. 

Does it make sense to keep it on the repo though? I was thinking about deploying the training (to gh pages or something like that) and maybe that's where the students can get the source `.svg` files from, and this repo can be more of a solution/showcase of custom challenges created by former trainees. What do you think?

File: src/app/solutions/belus.tsx [hunk 6]
```diff
@@ -218,9 +218,13 @@ export const BelusChallenge: React.FC<BelusChallengeProps> = ({}) => {
 
   return (
     <Animated.View style={[styles.flexible, bgStyle]}>
+      <Stack.Screen options={{ title: "Belu's Challenge" }} />
       <Animated.View style={[backgroundStyle, styles.background]} />
       <Animated.View style={fruitStyle}>
-        <Image source={Avocado} style={styles.fruit} />
+        <Image
+          source={require('../../../assets/images/food/avocado.png')}
```
Comment (carozo) [inline src/app/solutions/belus.tsx]: Just curious, why use `require` instead of `import`?

**Review by carozo — APPROVED** [review]
Looks great, thanks for contributing!

File: src/assets/mouth.svg (hunk 5 above)
Comment (lucasloisp) [reply src/assets/mouth.svg]: Answering both SVG threads here.

I love the idea of a "training guide" of sorts, hosted. I think it's possible to host a gh-pages from this same repo. We could even have a cli/script to create a "clean slate" version that has just the basic box for slimy, and perhaps some dependencies. Interesting backlog :smile: 

In the meantime, I'll bring back the svg files.

File: metro.config.js (hunk 3 above)
Comment (lucasloisp) [reply metro.config.js:9]: Answered on the other thread.

File: src/app/solutions/belus.tsx (hunk 6 above)
Comment (lucasloisp) [reply src/app/solutions/belus.tsx]: Good catch! Had some glitch on this, but it's working now - I was probably missing some path changes. Reverted!

**Comment by lucasloisp** [comment]
@carozo I just updated the branch, should be good to merge now!

File: src/assets/mouth.svg (hunk 5 above)
Comment (carozo) [reply src/assets/mouth.svg]: I love that idea! That way trainees can focus solely on animations 👏🏼

**Pull request merged**